sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -2122,4 +1912,242 @@ public void populateDisplayData(DisplayData.Builder builder) {
       config.populateDisplayData(builder);
     }
   }
+
+  /**
+   * {@link DoFn} class to write results data to a JDBC sink. It supports writing rows one by one
+   * (and returning individual results) - or by batch.
+   *
+   * @param <T>
+   * @param <V>
+   */
+  static class WriteFn<T, V> extends DoFn<T, V> {
```

Comment:
`AutoValue` requires that the enclosing class of the AutoValue class not be private, so I couldn't make this class private. However this class will be package private, so that should be pretty good right? : )